From 33455acfbe581812ac56f1d070183a933ad0f647 Mon Sep 17 00:00:00 2001
From: Jason Chen <b02280@freescale.com>
Date: Tue, 2 Aug 2011 15:43:18 +0800
Subject: [PATCH 0318/2463] ENGR00154135 sii902x hdmi: should not init twice

sii902x hdmi can only support one display, so second time init
function should return -EBUSY.

Signed-off-by: Jason Chen <b02280@freescale.com>
---
 drivers/video/mxc/mxcfb_sii902x.c |    8 +++++++-
 1 files changed, 7 insertions(+), 1 deletions(-)

diff --git a/drivers/video/mxc/mxcfb_sii902x.c b/drivers/video/mxc/mxcfb_sii902x.c
index 42d967f..02f798b 100644
--- a/drivers/video/mxc/mxcfb_sii902x.c
+++ b/drivers/video/mxc/mxcfb_sii902x.c
@@ -1067,6 +1067,12 @@ static int sii902x_disp_init(struct mxc_dispdrv_entry *disp)
 	struct mxc_dispdrv_setting *setting = mxc_dispdrv_getsetting(disp);
 	struct fsl_mxc_lcd_platform_data *plat = sii902x->client->dev.platform_data;
 	bool found = false;
+	static bool inited;
+
+	if (inited)
+		return -EBUSY;
+
+	inited = true;
 
 	setting->dev_id = plat->ipu_id;
 	setting->disp_id = plat->disp_id;
@@ -1139,7 +1145,7 @@ static int sii902x_disp_init(struct mxc_dispdrv_entry *disp)
 			mode = fb_find_nearest_mode(&m,
 					&sii902x->fbi->modelist);
 			fb_videomode_to_var(&sii902x->fbi->var, mode);
-			found = 1;
+			found = true;
 		}
 
 	}
-- 
1.7.7.4

